components/ParameterForm.vue

```diff
@@ -0,0 +1,236 @@
+            :label="parameter.label"
+            :aria-label="parameter.label"
+            :options="parameter.options.map((option: ParameterOption) => {
+              return { label: option.label, value: option.id };
+            })"
+            :size="largeScreen ? 'lg' : ''"
+          />
+        </div>
+      </div>
+      <div
+        v-if="globeParameter"
+        class="field-container"
+      >
+        <CIcon
+          v-if="icon(globeParameter)"
+          :icon="icon(globeParameter)"
+          class="parameter-icon"
+        />
+        <CFormSelect
+          :id="globeParameter.id"
+          v-model="formData[globeParameter.id]"
+          :data-test="`${globeParameter.id}-select`"
+          :label="globeParameter.label"
+          :aria-label="globeParameter.label"
+          :options="selectOptions(globeParameter)"
+          :size="largeScreen ? 'lg' : ''"
+        />
+      </div>
+      <CButton
+        id="run-button"
+        color="primary"
+        :size="largeScreen ? 'lg' : ''"
+        type="submit"
+      >
+        Run
+        <CIcon
+          icon="cilArrowRight"
+        />
+      </CButton>
+    </CForm>
+    <CAlert v-else-if="props.metadataFetchStatus === 'error'" color="warning">
+      Failed to retrieve metadata from R API. {{ metadataFetchError }}
+    </CAlert>
+    <CSpinner v-else-if="props.metadataFetchStatus === 'pending'" />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import type { FetchError } from "ofetch";
+import { CIcon } from "@coreui/icons-vue";
+import type { MetaData, Parameter, ParameterOption } from "@/types/daedalusApiResponseTypes";
+import type { AsyncDataRequestStatus } from "#app";
+
+const props = defineProps<{
+  globeParameter: Parameter | undefined
+  metaData: MetaData | undefined
+  metadataFetchStatus: AsyncDataRequestStatus
+  metadataFetchError: FetchError | null
+}>();
+
+const formData = ref(
+  // Create a new object with keys set to the id values of the metaData.parameters array of objects, and all values set to refs with default values.
+  props.metaData?.parameters.reduce((accumulator, parameter) => {
+    if (parameter.parameterType !== "select" && parameter.parameterType !== "globeSelect") {
+      accumulator[parameter.id] = ref("");
+      return accumulator;
+    }
+
+    // TODO: Make default country UK after November 2024 workshop
+    if (parameter.id === "country") {
+      accumulator[parameter.id] = ref("Thailand");
+    } else {
+      // Don't set an empty value or there will be a disjoint between the select component and the formData object,
+      // since the select component will visually appear to have an option selected (the first), but the formData object will have
+      // an empty string.
+      const defaultOption = parameter.defaultOption || parameter.options[0].id;
+      accumulator[parameter.id] = ref(defaultOption);
+    }
+
+    return accumulator;
+  }, {} as { [key: string]: any }),
```

Comment:
Turns out one of the 3rd-party components we're using, the radio button one, dictates that the type signature for the 'v-model' prop -- (which it translates into 'modelValue' behind the scenes!) -- is `string | boolean | unknown[] | undefined`.

From this, I will omit the Array type because we expect not to have multi-selects here.

I think I will keep the number type around as suggested, so it can be used by numeric inputs and we can do calculations with the value; and I'll just cast all values into strings when they are passed into the radio button v-model prop.